trin-storage/src/lib.rs

```diff
@@ -45,22 +45,18 @@ pub enum ShouldWeStoreContent {
 }
 
 /// A data store for Portal Network content (data).
-pub trait ContentStore {
+pub trait ContentStore<TContentKey: OverlayContentKey> {
```

Comment:
Something that I was thinking about before, and I think it would make sense to do it now (since you are refactoring anyway).

I believe that in the case of `ContentStore`, it might be better to use [Associated Type](https://doc.rust-lang.org/rust-by-example/generics/assoc_items/types.html), not the generic type. That way, the actual implementations (`HistoryStorage`/`BeaconStorage`/`StateStorage`) can specify the actual type and maybe cleanup code a bit as well. What do you think?

In the end, I don't think it makes big difference either way, so whatever you decide. But I think Associated type is better in this case?

Reply:
I gave this a go since it does seem cleaner, but I'm not sure if it's possible. Since `OverlayService` & `OverlayProtocol` are generic over the type of storage (`TStore`) and they use the `ContentStore` trait functions directly, then they need knowledge of this associated type. There's no way (from what I could find) to define the associated type on `ContentStore` as `impl OverlayContentKey`. So it seems to me like we're only able to define it with generics. Maybe I was doing something wrong/misunderstood how to implement this?

Reply:
Obviously, I'm not sure until I try it, but I believe that it would work the same way as with generics.
Meaning, if you named associated type `Key`, then instead of having: `ContentStore<TContentKey>`, you would have something like `ContentStore<Key=TContentKey>`. But I'm not sure if this is allowed because `TContentKey` is another generic...

Either way, feel free to leave it your way, and I will give it a try afterwards (if you agree that associated type is better).

Reply:
Ahh nice, yeah cool that syntax did the trick!